### PR TITLE
feat(DC): DC virtual_gateway and virtual_interface support tags

### PR DESCRIPTION
**Pull request by wanglong3710**
**What this PR does / why we need it**:
DC virtual_gateway and virtual_interface support tags

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.X
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dc/' TESTARGS='-run TestAccVirtualGateway_basic'
...
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/dc/ -v -run TestAccVirtualGateway_basic -timeout 360m -parallel 4 
=== RUN   TestAccVirtualGateway_basic 
=== PAUSE TestAccVirtualGateway_basic 
=== CONT  TestAccVirtualGateway_basic
--- PASS: TestAccVirtualGateway_basic (60.03s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dc        60.067s
```
```
make testacc TEST='./huaweicloud/services/acceptance/dc/' TESTARGS='-run TestAccVirtualInterface_basic'
...
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/dc/ -v -run TestAccVirtualInterface_basic -timeout 360m -parallel 4 
=== RUN   TestAccVirtualInterface_basic 
=== PAUSE TestAccVirtualInterface_basic
=== CONT  TestAccVirtualInterface_basic
--- PASS: TestAccVirtualInterface_basic (73.54s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dc        73.578s
```

### CODE REVIEW

File: huaweicloud/services/dc/resource_huaweicloud_dc_virtual_gateway.go [hunk 1]
```diff
@@ -118,6 +120,15 @@ func resourceVirtualGatewayCreate(ctx context.Context, d *schema.ResourceData, m
 	}
 	d.SetId(resp.ID)
 
+	// create tags
+	tagRaw := d.Get("tags").(map[string]interface{})
+	if len(tagRaw) > 0 {
```
Comment (deer-hang) [inline huaweicloud/services/dc/resource_huaweicloud_dc_virtual_gateway.go]: How about using `utils.CreateResourceTags`?

File: huaweicloud/services/dc/resource_huaweicloud_dc_virtual_gateway.go [hunk 2]
```diff
@@ -145,6 +156,17 @@ func resourceVirtualGatewayRead(_ context.Context, d *schema.ResourceData, meta
 		d.Set("status", resp.Status),
 	)
 
+	// Set tags
+	resourceTags, err := tags.Get(client, "dc-vgw", d.Id()).Extract()
```
Comment (deer-hang) [inline huaweicloud/services/dc/resource_huaweicloud_dc_virtual_gateway.go]: How about using `SetResourceTagsToState`?

File: docs/resources/dc_virtual_gateway.md [hunk 3]
```diff
@@ -53,6 +53,8 @@ The following arguments are supported:
   gateway belongs.  
   Changing this will create a new resource.
 
+* `tags` - (Optional, Map) The key/value pairs to associate with the virtual gateway.
```
Comment (ShiChangkuo) [inline docs/resources/dc_virtual_gateway.md]: please use `Specifies the ...` for parameters
```suggestion
* `tags` - (Optional, Map)  Specifies the key/value pairs to associate with the virtual gateway.
```

**Pull request merged**